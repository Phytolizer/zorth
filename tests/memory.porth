include "std.porth"

// stdout write "abc"
mem 0 + 97 .
mem 1 + 98 .
mem 2 + 99 .
mem 3 + 10 .

4 mem stdout write print

// increment chars
mem 0 + dup , 1 + .
mem 1 + dup , 1 + .
mem 2 + dup , 1 + .

// mem is now "bcd"
4 mem stdout write print

// UINT64_MAX
mem 18446744073709551615 .64
mem ,64 print
