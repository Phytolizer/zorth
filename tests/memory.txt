abc3
bcd3
