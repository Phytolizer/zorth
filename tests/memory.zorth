include "io.zorth"

mem 0 + 97 .
mem 1 + 98 .
mem 2 + 99 .
mem 3 + 10 .

4 mem write

mem 0 + dup , 1 + .
mem 1 + dup , 1 + .
mem 2 + dup , 1 + .

4 mem write
