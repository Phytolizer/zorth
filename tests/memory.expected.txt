: returncode 0
: stdout 33
abc
4
bcd
4
18446744073709551615

: stderr 0

